pkg/sync/sync_context.go

```diff
@@ -152,10 +152,11 @@ func WithResourceModificationChecker(enabled bool, diffResults *diff.DiffResultL
 }
 
 // WithNamespaceCreation will create non-exist namespace
-func WithNamespaceCreation(createNamespace bool, namespaceModifier func(*unstructured.Unstructured) bool) SyncOpt {
+func WithNamespaceCreation(createNamespace bool, nsMetaData map[string]interface{}, namespaceModifier func(*unstructured.Unstructured) bool) SyncOpt {
```

Comment:
Instead of accepting `map[string]interface{}`, could we just accept two parameters: `labels map[string]string, annotations map[string]string`? That would let us keep all the parsing logic in Argo CD.